### ISSUE TITLE
'/participant/connection' were called twice with connect  event

### ISSUE THREAD

**Issue by cythiaY** [amazon-connect/amazon-connect-chatjs#88]

**Comment by cythiaY** [comment]
Usage：
```
 const customerChatSession = connect.ChatSession.create({
      chatDetails: {
        contactId: ****,
        participantId: ****,
        participantToken: ****,
      },
      options: {
        region: "us-west-2",
      },
      type: connect.ChatSession.SessionTypes.CUSTOMER,
    });

    await customerChatSession.connect();
```

Network 👇:
![image](https://user-images.githubusercontent.com/32251309/177718519-d403a5bc-6c50-493c-95e3-1391e424185a.png)


Other，I can't pass “Type” parameter to function `connect` like `CreateParticipantConnection`

**Comment by mgerb001** [comment]
This should be two separate calls to the participant service. The first one is an OPTIONS method, the second the actual POST call.

**Comment by cythiaY** [comment]
Noop, both are POST method. And return different response.
first:
![image](https://user-images.githubusercontent.com/32251309/177723327-c3cf25b3-b9df-4b1d-b894-c812b4e8fb4b.png)

![image](https://user-images.githubusercontent.com/32251309/177723381-434a4111-1ab2-44db-be54-a84837449cb1.png)

second:
![image](https://user-images.githubusercontent.com/32251309/177723439-4217c424-bfed-47d9-bd72-16c4d0bc929a.png)

![image](https://user-images.githubusercontent.com/32251309/177723496-70a1f9fd-e4b3-46bf-aaef-c22e6c05dcaa.png)

**Comment by cythiaY** [comment]
Can anyone help me？😣

**Comment by haomingli2020** [comment]
Hi, do you know if this is a new issue in the latest version or was it existing in old versions as well?

**Comment by sseidel16** [comment]
This behavior is ongoing for some time. Possibly since the beginning of the library. There was a release awhile back that was designed to eliminate the double calls, but it was rolled back in the next release because of some issues that I do not have details on. The release that addressed this was https://github.com/amazon-connect/amazon-connect-chatjs/tree/1.0.7, then it was rolled back.

**Comment by xinyu-shopastro** [comment]
> This behavior is ongoing for some time. Possibly since the beginning of the library. There was a release awhile back that was designed to eliminate the double calls, but it was rolled back in the next release because of some issues that I do not have details on. The release that addressed this was https://github.com/amazon-connect/amazon-connect-chatjs/tree/1.0.7, then it was rolled back.

Verison 1.0.7 doesn't work, even without TS.
When will u fix this problem in the future? 😊

**Comment by cythiaY** [comment]
> This behavior is ongoing for some time. Possibly since the beginning of the library. There was a release awhile back that was designed to eliminate the double calls, but it was rolled back in the next release because of some issues that I do not have details on. The release that addressed this was https://github.com/amazon-connect/amazon-connect-chatjs/tree/1.0.7, then it was rolled back.

Verison 1.0.7 doesn't work, even without TS.
When will u fix this problem in the future? 😊

**Comment by haomingli2020** [comment]
We have the item to fix it in our backlog but it hasn't been prioritized yet. Can you explain more on how this behavior becomes a problem for you?

**Comment by cythiaY** [comment]
It's not urgent now. 
We just listen to the ’connect‘ event in server to do something. We have found another workaround.

**Comment by zhangyuezhong** [comment]
I am see the same behaviour,  the connection is called twice,   each time it turn a different websocket url(token is different),  the library use the 2nd response token to establish the websocket.

**Comment by zhangyuezhong** [comment]
> 
I believe the problem probably related to the connectionDetailsProvider.js 


baseConnectionHelper.js (line 32)  ---> invokes  this.connectionDetailsProvider.fetchConnectionToken()
LpcConnectionHelper.js  (line 139) ----> invokes the connectionDetailsProvider.fetchConnectionDetails()

these calls will cause the provide to _callCreateParticipantConnection twice (and create two participant connections). 

the   fetchConnectionDetails() and fetchConnectionToken() should check if connectionDetails or token already available, if not make a connection otherwise return it. 

![image](https://user-images.githubusercontent.com/23203638/190660789-49629016-03ac-42af-8ff8-cd0096ac839c.png)

**Comment by marcogrcr** [comment]
This may be related to #31.

**Comment by mhiaror** [comment]
This issue is fixed in the latest release 1.3.2
